contracts/root/TokenPredicates/ITokenPredicate.sol

```diff
@@ -2,18 +2,36 @@ pragma solidity ^0.6.6;
 
 import {RLPReader} from "../../lib/RLPReader.sol";
 
+/// @title Token predicate interface for all pos portal prdicates
+/// @notice Abstract interface that defines methods for custom predicates
 interface ITokenPredicate {
+
+    /// @notice Deposit tokens into pos portal
+    /// @dev When `depositor` deposits tokens into pos portal, tokens get locked into predicate contract.
+    /// @param depositor Address who wants to deposit tokens
+    /// @param depositReceiver Address (address) who wants to receive tokens on side chain
+    /// @param rootToken Token which gets deposited
+    /// @param depositData Extra data for deposit (amount for ERC20, token id for ERC721 etc.) [ABI encoded]
     function lockTokens(
         address depositor,
         address depositReceiver,
         address rootToken,
         bytes calldata depositData
     ) external;
 
+    /// @notice Validates exit while withdraw process
+    /// @dev Validates exit log while withdrawing. Reverts if validation fails
+    /// @param withdrawer Address who wants to withdraw tokens
+    /// @param logRLPList Log bytes from sidechain
     function validateExitLog(address withdrawer, bytes calldata logRLPList)
```

Comment:
I spoke to Arth earlier about removing this. So let #15 be merged first - assuming this change is present in that PR.

Reply:
yep removing `lockTokens` and moving its logic to `exitTokens` as part of https://github.com/maticnetwork/pos-portal/pull/15

Reply:
changed this